app/src/sharedTest/java/org/oppia/app/recyclerview/BindableAdapterTest.kt

```diff
@@ -251,6 +251,62 @@ class BindableAdapterTest {
     }
   }
 
+  @Test
+  fun testBindableAdapter_incomingSameData_noRebindingShouldHappen() {
+
+    val adapter = createMultiViewTypeNoDataBindingBindableAdapter()
+    BindableAdapterTestFragmentPresenter.testBindableAdapter = adapter
+
+    lateinit var dataObserver: RecyclerView.AdapterDataObserver
+    var doesRebind = false
+
+    val oldList = listOf(STR_VALUE_1, STR_VALUE_1, INT_VALUE_1).toMutableList()
+    val newList = listOf(STR_VALUE_1, STR_VALUE_1, INT_VALUE_1).toMutableList()
+
+    ActivityScenario.launch(BindableAdapterTestActivity::class.java).use { scenario ->
+
+      scenario.onActivity { activity ->
+        dataObserver = object : RecyclerView.AdapterDataObserver() {
+          override fun onChanged() {
+            doesRebind = true
+          }
+
+          override fun onItemRangeInserted(positionStart: Int, itemCount: Int) {
+            assertThat(positionStart).isEqualTo(0)
```

Comment:
Can we extract this state & assert in the test body, instead? Prefer keeping tests with the structure: arrange, act, assert. We can collect data during the 'act' step to use in 'assert', but we should keep these two things separate.

Reply:
updated test case